torch/_jit_internal.py

```diff
@@ -903,6 +903,8 @@ def _qualified_name(obj):
     # Enum classes do not have `__name__` attr, instead they have `name`.
     elif isinstance(obj, enum.Enum):
         name = obj.name
+    elif hasattr(obj, '__str__'):
+        return obj.__str__()
```

Comment:
I think TorchScript uses qualified name as indexing key in various places. and `__str__` could return anything that is not guaranteed to be unique. Could you help me understand why choosing `__str__` as the name rather than something like `id()`?

Reply:
I wasn't sure  __str__ doesn't return a unique output, I will update the code to use id

Reply:
`__str__` can be user-defined, so that's why there's no guarantee that it returns a unique output.